test/test_utils.py

```diff
@@ -136,6 +137,93 @@ def test_vcap_credentials():
     assert authenticator is None
     del os.environ['VCAP_SERVICES']
 
+    vcap_services = '{"test":[{"name": "testname",\
+        "credentials":{ \
```

Comment:
In order to simplify the tests and make it easy to add new vcap-related tests in the future, I think it would be good to capture all of the VCAP_SERVICES testdata perhaps in a file, then just read in that file and set the VCAP_SERVICES env var to contain the contents of the file, similar to the Go core tests.
In fact, you could probably just copy the same file and check it into the python core.

Reply:
I think it might be simplier to leave them as json snippets in the code like this since each one is slight different for each different test. If we put them in a file we'd need to lable each one to each test they go to and parse them from the file correct?